### PR TITLE
DAO in progress

### PR DESCRIPTION
**Pull request by JK8789**

### CODE REVIEW

File: PetFeeding/pom.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.example</groupId>
+    <artifactId>PetFeeding</artifactId>
```
Comment (DanilaVaratyntsev) [inline PetFeeding/pom.xml]: it's usually lowercase. But I don't insist

File: PetFeeding/src/main/java/com/epam/ConnectionSource.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.stream.Collectors;
+
+public class ConnectionSource {
+    // JDBC driver name and database URL
+    static final String JDBC_DRIVER = "com.mysql.jdbc.Driver";
+    private static final String DB_URL = "jdbc:mysql://localhost:3306/petsdb";
+
+    //  Database credentials
+    private static final String USER = "root";
+    private static final String PASS = "password";
```
Comment (DanilaVaratyntsev) [inline PetFeeding/src/main/java/com/epam/ConnectionSource.java]: Don't hardcode passwords, use properties file that is not committed to git

File: PetFeeding/src/main/java/com/epam/dao/Dao.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+public interface Dao<T, Id> {
+
+    Optional<T> getById(Id Id);
+
+    List<T> getAll();
+
+    T save(T t);
+
+    void delete(T t);
+}
```
Comment (DanilaVaratyntsev) [inline PetFeeding/src/main/java/com/epam/dao/Dao.java]: Please add newlines to the end of all files. Some tools require it

File: PetFeeding/src/main/java/com/epam/dao/DaoFactory.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    private Connection connection;
+
+    public DaoFactory() throws SQLException {
+        connectionSource = ConnectionSource.instance();
+        connection = connectionSource.createConnection();
+    }
+
+//    public EmployeeDao employeeDAO() throws SQLException {
+//        EmployeeDao employeeDao = new EmployeeDaoImpl(connection);
+//        return employeeDao;
```
Comment (DanilaVaratyntsev) [inline PetFeeding/src/main/java/com/epam/dao/DaoFactory.java]: DAOs should be singletons, too

File: PetFeeding/src/main/java/com/epam/dao/PersonDao.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.epam.dao;
+
+import com.epam.domain.Person;
+
+import java.math.BigInteger;
+
+public interface PersonDao extends Dao<Person, BigInteger> {
```
Comment (DanilaVaratyntsev) [inline PetFeeding/src/main/java/com/epam/dao/PersonDao.java]: Integer or Long is enough, you don't need BigInt

**Review by DanilaVaratyntsev — APPROVED** [review]

File: PetFeeding/pom.xml (hunk 1 above)
Comment (JK8789) [reply PetFeeding/pom.xml]: fixed

File: PetFeeding/src/main/java/com/epam/ConnectionSource.java (hunk 2 above)
Comment (JK8789) [reply PetFeeding/src/main/java/com/epam/ConnectionSource.java]: fixed

File: PetFeeding/src/main/java/com/epam/dao/Dao.java (hunk 3 above)
Comment (JK8789) [reply PetFeeding/src/main/java/com/epam/dao/Dao.java]: fixed

File: PetFeeding/src/main/java/com/epam/dao/PersonDao.java (hunk 5 above)
Comment (JK8789) [reply PetFeeding/src/main/java/com/epam/dao/PersonDao.java]: fixed

**Pull request merged**